homeassistant/components/justnimbus/sensor.py

```diff
@@ -0,0 +1,199 @@
+        value_fn=lambda coordinator: coordinator.data.reservoir_content,
+    ),
+    JustNimbusEntityDescription(
+        key="total_saved",
+        name="Total saved",
+        icon="mdi:water-opacity",
+        native_unit_of_measurement=VOLUME_LITERS,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda coordinator: coordinator.data.total_saved,
+    ),
+    JustNimbusEntityDescription(
+        key="total_replenished",
+        name="Total replenished",
+        icon="mdi:water",
+        native_unit_of_measurement=VOLUME_LITERS,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda coordinator: coordinator.data.total_replenished,
+    ),
+    JustNimbusEntityDescription(
+        key="error_code",
+        name="Error code",
+        icon="mdi:bug",
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement="",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda coordinator: coordinator.data.error_code,
+    ),
+    JustNimbusEntityDescription(
+        key="totver",
+        name="Total use",
+        icon="mdi:chart-donut",
+        native_unit_of_measurement=VOLUME_LITERS,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda coordinator: coordinator.data.totver,
+    ),
+    JustNimbusEntityDescription(
+        key="reservoir_content_max",
+        name="Max reservoir content",
+        icon="mdi:waves",
+        native_unit_of_measurement=VOLUME_LITERS,
+        state_class=SensorStateClass.TOTAL,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda coordinator: coordinator.data.reservoir_content_max,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the JustNimbus sensor."""
+    coordinator: JustNimbusCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        JustNimbusSensor(
+            device_id=entry.data[CONF_CLIENT_ID],
+            description=description,
+            coordinator=coordinator,
+        )
+        for description in SENSOR_TYPES
+    )
+
+
+class JustNimbusSensor(
+    JustNimbusEntity,
+):
+    """Implementation of the JustNimbus sensor."""
+
+    def __init__(
+        self,
+        *,
+        device_id: str,
+        description: JustNimbusEntityDescription,
+        coordinator: JustNimbusCoordinator,
+    ) -> None:
+        """Initialize the sensor."""
+        self.entity_description: JustNimbusEntityDescription = description
+        super().__init__(
+            device_id=device_id,
+            coordinator=coordinator,
+        )
+        self._device_id = device_id
+        self._attr_unique_id = f"{coordinator.entry_id}_{description.key}"
```

Comment:
We can use device ID instead here, right?
That removes the need for juggling with the config entry.

```suggestion
        self._attr_unique_id = f"{device_id}_{description.key}"
```